docs/rules/expiring-todo-comments.md

```diff
@@ -0,0 +1,108 @@
+# Add conditions to TODO comments to make them error
+
+Makes it possible to pass arguments to TODO and FIXME comments to trigger errors.
+
+This rule also implements [`eslint/no-warning-comments`](https://eslint.org/docs/rules/no-warning-comments) behavior so you should disable that in favor of using this.
+
+For projects with legacy branches, such as long term supported older versions, there might be situations where you fix TODOs in the master branch but on the other branches you leave it missing. Choose carefully which projects you would want this feature.
+
+Supported conditions:
+
+- `[YYYY-MM-DD]` to define a due date.
+- `[>1]` or `[>=2]` to expire at some version (from package.json). No whitespace.
+- `[+package]` or `[-package]` to expire when you add/remove a package.
+- `[package@>1]` or `[package@>=2]` to expire when a package hits some version. No whitespace.
+- `[engines node>8]` or `[engines node>=8]` to expire when package bump supported engines. Don't use spaces between the comparison.
+
+Info:
+- Multiple conditions can be separated by comma: `[YYYY-MM-DD, +foo-package, engines node>=8, bar-package@>=2]`.
```

Comment:
Don't forget to keep the docs in sync with changes.

Reply:
Sure. I was wrapping up code before going for the docs.

Reply:
Done

Reply:
There are some Grammer mistakes but will wait for @sindresorhus to review first.

Reply:
I tried to use a grammar checking tool but I knew we'd miss some, sorry 😋